.github/workflows/ci-cd-workflow.yml

```diff
@@ -0,0 +1,81 @@
+name: elba CI-CD
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*'
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: ["ubuntu-latest"]
+        python-version: [3.7, 3.8, 3.9, 3.10, 3.11]
+        sphinx-version: ["<6", "<7", ">=6"]
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+    - name: Setup python
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install test dependencies
+      run: |
+        pip install --upgrade --user pip wheel
+        pip install .[tests] 'sphinx ${{ matrix.sphinx-version }}'
+        pip freeze
+    - name: Test with pytest (${{ runner.os }}), python ${{ matrix.python-version }}, sphinx ${{ matrix.sphinx-version }}
+      run: |
+          pytest tests -r a
+
+  create-dist:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.9]
```

Comment:
I think we want to package on the oldest python we support?  Does it even matter for a pure python lib?

Reply:
> Does it even matter for a pure python lib?

I have no idea (intuitively, I doubt it...).